src/main/java/hudson/plugins/accurev/AccurevSCM.java

```diff
@@ -366,14 +364,22 @@ public void buildEnvironment(Run<?, ?> build, Map<String, String> env) {
      * @throws java.lang.InterruptedException on failing interrupt
      */
 
-    public void checkout(@Nonnull Run<?, ?> build, @Nonnull Launcher launcher, @Nonnull FilePath workspace,
-                         @Nonnull TaskListener listener, @CheckForNull File changelogFile,
-                         @CheckForNull SCMRevisionState baseline) throws IOException, InterruptedException {
-//        TODO: Implement SCMRevisionState?
-        boolean checkout = AccurevMode.findDelegate(this).checkout(build, launcher, workspace, listener, changelogFile);
-        if (checkout) listener.getLogger().println("Checkout done");
-        else listener.getLogger().println("Checkout failed");
-    }
+	public void checkout(@Nonnull Run<?, ?> build, @Nonnull Launcher launcher, @Nonnull FilePath workspace,
+			@Nonnull TaskListener listener, @CheckForNull File changelogFile, @CheckForNull SCMRevisionState baseline)
+			throws IOException, InterruptedException {
+		 AccurevServer server = getServer();
+		 boolean serverDisabled = server != null && server.isServerDisabled();
+		if (!serverDisabled) {
+			boolean checkout = AccurevMode.findDelegate(this).checkout(build, launcher, workspace, listener,
+					changelogFile);
+			if (checkout)
+				listener.getLogger().println("Checkout done");
+			else
+				listener.getLogger().println("Checkout failed");
+		} else {
+			listener.getLogger().println("Checkout skipped");
```

Comment:
This won't work... now your just breaking builds without proper checkout. Please make it throw `AbortException`.

Reply:
Please change it to the something like this instead.
```java
AccurevServer server = getServer();
boolean serverDisabled = server != null && server.isServerDisabled();
if (serverDisabled) {
  throw new AbortException("Checkout skipped");
}
```